### PR TITLE
EP-69: add support for separate logging configurations by log type

### PR DESCRIPTION
**Pull request by codeSonic**

### CODE REVIEW

File: lib/plugins/sink/firehose/firehoseSink.js [hunk 1]
```diff
@@ -37,30 +40,14 @@ class FirehoseSink extends ILogSink {
...
-        }
-
-        this[_streamNames] = streamSetting;
-        this[_firehosePool] = Firehose(getPoolName(accessKeyId, secretAccessKey, region), {
-            accessKeyId: accessKeyId,
-            secretAccessKey: secretAccessKey,
-            region: region,
-            max_clients: (!isNaN(options.max_clients) && options.max_clients > 0) ? options.max_clients : 30
-        });
+        _typedSink["api"] = getSink(firehoseLogSettings.apiLogSettings, options);
```
Comment (apalsapure) [inline lib/plugins/sink/firehose/firehoseSink.js]: Please add check for `firehoseLogSettings` for null

File: lib/plugins/sink/firehose/firehoseSink.js [hunk 2]
```diff
@@ -7,21 +7,24 @@ var ILogSink = require('../../../modules/logging/iLogSink');
...
-var getStreamName = function (log, streamNames) {
-    var logType = log['type'];
-    var streamName = `${logType}_stream`;
-    if (logType && streamNames[streamName])
-        return streamNames[streamName];
-    else
-        return streamNames['stream'];
-};
+var getSink = function(firehoseSettings, options){
+    var firehosePool = Firehose(getPoolName(firehoseSettings.accessKeyId, firehoseSettings.secretAccessKey, firehoseSettings.region), {
```
Comment (apalsapure) [inline lib/plugins/sink/firehose/firehoseSink.js]: Please add check for `firehoseSettings`

File: lib/plugins/sink/redis/redisSink.js [hunk 3]
```diff
@@ -29,6 +28,31 @@ var isIpAddress = function (host) {
     return result;
 };
 
+var getSink = function(redisSettings, options){
+    if (typeof (redisSettings.queueName) !== 'string')
```
Comment (apalsapure) [inline lib/plugins/sink/redis/redisSink.js]: Please add check for `redisSettings`

File: lib/plugins/sink/redis/redisSink.js [hunk 4]
```diff
@@ -41,38 +65,17 @@ class RedisSink extends ILogSink {
...
-        var cfg = {
-            host: host,
-            port: port,
-            max_clients: (!isNaN(options.max_clients) && options.max_clients > 0) ? options.max_clients : 30
-        };
-        if(options.sslEnabled)
-            cfg.options = { tls: { servername: host } };
-        this[_redisPool] = RedisPool(getPoolName(port, host), cfg);
-        this[_queueName] = queues;
+        _typedSink["api"] = getSink(redisLogSettings.apiLogSettings, options);
```
Comment (apalsapure) [inline lib/plugins/sink/redis/redisSink.js]: Please add check for `redisLogSettings`

**Comment by codeSonic** [comment]
verify please

**Pull request merged**